lib/plug/telemetry.ex

```diff
@@ -0,0 +1,26 @@
+defmodule Plug.Telemetry do
+  @moduledoc """
+  A plug to instrument the pipeline with `:telemetry` events.
+  """
+
+  @behaviour Plug
+
+  @impl true
+  def init(_), do: []
+
+  @impl true
+  def call(conn, _opts) do
+    time = System.monotonic_time()
+    :telemetry.execute([:plug, :call, :start], %{}, %{conn: conn})
```

Comment:
If these happen synchronously (which I think is the case for Telemetry?), then the subscriber could choose to check the timestamp itself, or perhaps take the `monatonic_time` value and format it as needed. My concern with that is that we might end up in a similar place to how [Cowboy manages its own cache of pre-formatted timestamps for efficiency](https://github.com/ninenines/cowboy/blob/master/src/cowboy_clock.erl), and it's unclear to me where it makes sense to put something like that, so that everyone benefits from the same cache.

Reply:
They could do it themselves, but I've found that generally you need both `System.system_time()` and `System.monotonic_time()`. 

The first marks the actual start time, and the monotonic time is required to calculate accurate duration. You never need to format the monotonic time, since it's only used in calculating duration

FYI for those interested, time is documented _very_ thoroughly in https://learnyousomeerlang.com/time and the tldr; for how to use them is https://learnyousomeerlang.com/time#how-to-survive-time-warps

Reply:
@arkgil because in Telemetry.Metrics we will recommend checking if the measurement is there, I think we cannot pass empty measurements. Therefore I think passing `time: System.system_time` as the measurement is a good call.